### DISCUSSION TITLE
Syntax error in generated cache file

### DISCUSSION THREAD

**Discussion by Nyde** [phalcon/cphalcon#16114] [Q&A] [answered]
Good morning,

after upgrade to 5.0.0 on php 8.1.6 I'm working through the migration; now I stumbled across this:

```
 Parse error: syntax error, unexpected token "=>", expecting ")" in C:\dev\xampp816\htdocs\cache\c__dev_xampp816_htdocs_app_views_index_index.volt.cache on line 34
```

The line in question is:
```php
<?= $this->tag->form("/", 'method' => "post", 'role' => "form", 'name' => "loginForm") ?>
```

Deleting the cache file and recreating it yields the same.

Config goes like the usual in 4.1:

```php
$view->registerEngines([
        '.volt' => function ($view) {
            $config = $this->getConfig();

            $volt = new VoltEngine($view, $this);

            $options = [
                'path' => $config->application->cacheDir,
                'separator' => '_',
                'extension' => '.cache'
            ];

            if($config->application->enviroment == 'dev') // which is set actually, but it still creates the cache files
                $options['always'] = true;

            $volt->setOptions($options);

            return $volt;
        },
        '.phtml' => PhpEngine::class

    ]);
```

Did 5.0.0 change anything of these?

Thanks.

**Comment by Ultimater** [comment]
What's your volt look like for that line and what did you upgrade from where it used to work?

**Reply by Nyde** [reply]
The corresponding line in Volt is:
```volt
{{ form('/', 'method': 'post', 'role': 'form', 'name': 'loginForm') }}
```

So far, I only changed the new namespace structure, e.g. Phalcon\Loader to Phalcon\Autoload\Loader etc.

**Comment by Nyde** [comment, marked as answer]
Ok, I just noticed there is a new syntax for the form arguments, with ```[]``` enclosing the values and a action key; wasn't on the upgrading page, but I found it on the 5.0.0 Volt docs.

**Reply by niden** [reply]
@Nyde you are using the `form()` method that is generated by `Phalcon\Tag`. The `form()` in v5 is generated by `Phalcon\Html\TagFactory` which has a different signature.

You can either change the signature and use `form()` or change to `formLegacy()` which will use the old `Phalcon\Tag` (and not change the signature)

**Reply by niden** [reply]
I will add a notice in the upgrade docs also
